src/compiler/utilitiesPublic.ts

```diff
@@ -258,8 +258,13 @@ namespace ts {
     }
 
     export type ParameterPropertyDeclaration = ParameterDeclaration & { parent: ConstructorDeclaration, name: Identifier };
+
     export function isParameterPropertyDeclaration(node: Node, parent: Node): node is ParameterPropertyDeclaration {
-        return hasSyntacticModifier(node, ModifierFlags.ParameterPropertyModifier) && parent.kind === SyntaxKind.Constructor;
+        return isParameter(node) &&
+            !!node.name && // NOTE: JSDoc parameters may not have names

```

Comment:
I can't recall. _something_ broke that triggered me to change this while I was working on the `ts` transform, but that was weeks ago. I'll have to revert it to see what broke.

Reply:
Reverting this doesn't seem to break anything. I think I fixed it to work around a crash in a test with an invalid `constructor(public this)` parameter, but have since changed some of the logic in checker so that decorators on such parameters are ignored, so the crash is no longer triggered. I've reverted this change.